x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -338,19 +338,31 @@ public void testInlineStatsWithoutGroups() {
         );
     }
 
-    public void testIdentifiersAsIndexPattern() {
-        // assertIdentifierAsIndexPattern("foo", "from `foo`");
-        // assertIdentifierAsIndexPattern("foo,test-*", "from `foo`,`test-*`");
-        assertIdentifierAsIndexPattern("foo,test-*", "from foo,test-*");
-        assertIdentifierAsIndexPattern("123-test@foo_bar+baz1", "from 123-test@foo_bar+baz1");
-        // assertIdentifierAsIndexPattern("foo,test-*,abc", "from `foo`,`test-*`,abc");
-        // assertIdentifierAsIndexPattern("foo, test-*, abc, xyz", "from `foo, test-*, abc, xyz`");
-        // assertIdentifierAsIndexPattern("foo, test-*, abc, xyz,test123", "from `foo, test-*, abc, xyz`, test123");
-        assertIdentifierAsIndexPattern("foo,test,xyz", "from foo,   test,xyz");
-        assertIdentifierAsIndexPattern(
-            "<logstash-{now/M{yyyy.MM}}>", // ,<logstash-{now/d{yyyy.MM.dd|+12:00}}>
-            "from <logstash-{now/M{yyyy.MM}}>" // , `<logstash-{now/d{yyyy.MM.dd|+12:00}}>`
+    public void testStringAsIndexPattern() {
+        assertStringAsIndexPattern("foo", "from \"foo\"");
+        assertStringAsIndexPattern("foo,test-*", """
+            from "foo","test-*"
+            """);
+        assertStringAsIndexPattern("foo,test-*", "from foo,test-*");
+        assertStringAsIndexPattern("123-test@foo_bar+baz1", "from 123-test@foo_bar+baz1");
+        assertStringAsIndexPattern("foo,test-*,abc", """
+            from "foo","test-*",abc
+            """);
+        assertStringAsIndexPattern("foo, test-*, abc, xyz", """
+                from "foo, test-*, abc, xyz"
+            """);
+        assertStringAsIndexPattern("foo, test-*, abc, xyz,test123", """
+                from "foo, test-*, abc, xyz", test123
+            """);
+        assertStringAsIndexPattern("foo,test,xyz", "from foo,   test,xyz");
+        assertStringAsIndexPattern(
+            "<logstash-{now/M{yyyy.MM}}>,<logstash-{now/d{yyyy.MM.dd|+12:00}}>",
```

Comment:
Suggestions for additional test cases:
- Maybe let's also add a simple test case without quotes but with colons `:` to illustrate that CCQ is fine.
```
from logs-*,*:logs-*
```
- We could also add a test case with an index name containing and/or starting with a dot `.`, especially without quotes.
- What happens when quotes are not properly closed, or opened? We could add a small randomized test for `"FROM " + leftQuote + "some-idx" + rightQuote`, where the left and right quotes are a random, mismatched selection from `", """` and the empty string (unquoted).
- Other pathological cases:
  - what if a quoted string contains an un-escaped quote in the middle? E.g. `FROM """someidx"""name"""` or `FROM "someidx"name"`.
  - What if two quoted strings are next to each other? `FROM "someidx""name"`

Reply:
`from "metadata"` would also help.